ext/node/polyfills/http2.ts

```diff
@@ -357,7 +428,8 @@ function closeSession(session: Http2Session, code?: number, error?: Error) {
 
 export class ServerHttp2Session extends Http2Session {
   constructor() {
-    super(constants.NGHTTP2_SESSION_SERVER, {});
+    // TODO(satyarohith): pass socket instead of undefined
```

Comment:
I'm not fully clear on the semantics of Deno <=> Rust ownership, so I don't know if this is easily shared. But if it is, I can just make the change lower in this file:

```diff
     this.on(
       "connection",
       (conn: Deno.Conn) => {
         try {
-          const session = new ServerHttp2Session();
+          const session = new ServerHttp2Session(conn);
           this.emit("session", session);
           this.#server = serveHttpOnConnection(
             conn,
```

Reply:
This should be okay to do; "Deno.Conn" is a resource backed by a Rust resource. If the Rust resource is moved the worst we get is `BadResource` error in JS, which seems preferable than not having connection available on the session object at all.

Reply:
I spoke too eagerly, it seems there isn't a straight forward way to create a `net.Socket` from a `Deno.Conn`, or if there is I haven't found it.

Reply:
Yes. We have to refactor `net.Socket` to work on top of rust resource that backs `Deno.Conn`. We can do that in a future iteration.